Civi/Payment/PropertyBag.php

```diff
@@ -38,12 +38,14 @@ class PropertyBag implements \ArrayAccess {
     'billingPostalCode'           => TRUE,
     'billingCounty'               => TRUE,
     'billingCountry'              => TRUE,
+    'cardNumber'                  => TRUE,
     'contactID'                   => TRUE,
     'contact_id'                  => 'contactID',
     'contributionID'              => TRUE,
     'contribution_id'             => 'contributionID',
     'contributionRecurID'         => TRUE,
     'contribution_recur_id'       => 'contributionRecurID',
+    'credit_card_number'          => 'cardNumber',
```

Comment:
In fact this should be TRUE & cardNumber should be a convenience getter for this or debit_card_number to honour the existing way it is passed in

Reply:
(In all the other discussion I missed that this PR wasn't doing what I intended - ie expose the existing expectations about what fields a payment processor receives)

Reply:
as in
```
'credit_card_number'          => 'creditCardNumber',
' 'creditCardNumber'' => TRUE,

```

?

Reply:
@artfulrobot I think this PR does that - but actually I think cardNumber should be creditCardNumber, leaving space for debitCardNumber (cardNumber could then choose the right one).

Reply:
@artfulrobot yeah - I'm not wedded to the cardNumber helper being present.

Reply:
oh I see map both of them to it - I wonder if that has any advantage over just creditCardNumber & debitCardNumber (which are the field names we offer up)

Reply:
Hmm - the metadata in core for those fields has been standard for a  long time. The fact they aren't necessarily the same length seems pretty tangental. We do currently validate that Amex has 4 digit cvvs vs other cards having 3 - regardless of which payment processor is involved - because it's not processor specific.